process/headerCheck/headerSignatureVerify.go

```diff
@@ -258,9 +264,20 @@ func (hsv *HeaderSigVerifier) getPrevHeaderInfo(currentHeader data.HeaderHandler
 
 // VerifyPreviousBlockProof verifies if the structure of the header matches the expected structure in regards with the consensus flag
 func (hsv *HeaderSigVerifier) VerifyPreviousBlockProof(header data.HeaderHandler) error {
-	previousAggregatedSignature, previousBitmap := header.GetPreviousAggregatedSignatureAndBitmap()
-	hasProof := len(previousAggregatedSignature) > 0 && len(previousBitmap) > 0
-	hasLeaderSignature := len(previousBitmap) > 0 && previousBitmap[0]&1 != 0
+	previousProof := header.GetPreviousProof()
+
+	hasProof := false
+	hasLeaderSignature := false
+
+	if previousProof != nil {
+		previousAggregatedSignature, previousBitmap := previousProof.GetAggregatedSignature(), previousProof.GetPubKeysBitmap()
+		hasProof = len(previousAggregatedSignature) > 0 && len(previousBitmap) > 0
+
+		if len(previousBitmap) > 0 {
+			hasLeaderSignature = previousBitmap[0]&1 != 0
```

Comment:
this is no longer valid
with fixed ordering, the leader can have any index in the bitmap, but at the same time it is no longer needed to have its signature in the proof.